File: builder/nddev-builder/hooks/nddev-builder-pretooluse.mjs
#!/usr/bin/env node
const chunks = [];
process.stdin.setEncoding("utf8");
process.stdin.on("data", (chunk) => chunks.push(chunk));
process.stdin.on("end", () => {
  let payload = {};
  try {
    payload = JSON.parse(chunks.join("") || "{}");
  } catch {
    process.exit(0);
  }
  const command = String(payload.command || payload.tool_input?.command || "");
  if (/\brm\s+-rf\b/.test(command) || /^\s*sudo\b/.test(command)) {
    console.error("nddev-builder blocks destructive shell command patterns in managed setups");
    process.exit(2);
  }
  process.exit(0);
});

File: builder/nddev-builder/hooks/nddev-builder-pretooluse.py
#!/usr/bin/env python3
"""Kimi PreToolUse hook for NDDev target-owned setup state."""

from __future__ import annotations

import json
import re
import sys


def main() -> int:
    try:
        payload = json.load(sys.stdin)
    except json.JSONDecodeError:
        return 0
    if not isinstance(payload, dict):
        return 0
    tool_input = payload.get("tool_input")
    command = ""
    if isinstance(tool_input, dict):
        command = str(tool_input.get("command") or "")
    if not command:
        command = str(payload.get("command") or "")
    if re.search(r"\brm\s+-rf\s+/(?:\s|$)", command) or re.match(r"\s*sudo(?:\s|$)", command):
        print("nddev-builder blocks destructive shell command patterns in managed setups", file=sys.stderr)
        return 2
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
